sqlglot/transforms.py

```diff
@@ -154,7 +154,8 @@ def remove_precision_parameterized_types(expression: exp.Expression) -> exp.Expr
     """
     for node in expression.find_all(exp.DataType):
         node.set(
-            "expressions", [e for e in node.expressions if not isinstance(e, exp.DataTypeParam)]
```

Comment:
there are a lot of unnecessary changes, can you revert them all?

Reply:
reverted